web/Services/CsvImportService.cs

```diff
@@ -0,0 +1,133 @@
+    {
+        _logger = logger;
+        _dbContext = dbContext;
+    }
+
+    public (List<string>, List<string>) CheckIfImportWorks(IFormFile csvFile)
+    {
+        List<string> flashTypes = new(),
+            flashMessages = new();
+        var headerLine = "";
+        using (var streamReader = new StreamReader(csvFile.OpenReadStream()))
+        {
+            headerLine = streamReader.ReadLine();
+        }
+        
+        var headers = headerLine!.Split(",");
+        var databaseHeaders = new List<string>();
+        
+        //check for missing columns in csv file
+        foreach (var dataProp in typeof(TenantUploadModel).GetProperties())
+        {
+            if (!headers.Contains(dataProp.Name) && dataProp.Name != "Id")
+            {
+                _logger.LogWarning($"Missing column: {dataProp.Name}");
+                flashTypes.Add("danger");
+                flashMessages.Add(
+                    $"Import has failed because column {dataProp.Name} is missing. Please add this column to your records before attempting to import them.");
+            }
+            databaseHeaders.Add(dataProp.Name);
+        }
+
+        //check for extra columns in csv file
+        foreach (var header in headers)
+        {
+            if (!databaseHeaders.Contains(header))
+            {
+                _logger.LogWarning($"Extra column: {header}");
+                flashTypes.Add("warning");
+                flashMessages.Add(
+                    $"The column {header} does not exist in the database. All data in this column has been ignored.");
+            }
+        }
+        return (flashTypes, flashMessages);
+    }
+
+    public async Task<(List<string> FlashTypes, List<string> FlashMessages)> ImportTenants(IFormFile csvFile, (List<string> flashTypes, List<string> flashMessages) flashResponse)
+    {
+        CsvFileDescription csvFileDescription = new CsvFileDescription
+        {
+            SeparatorChar = ',',
+            FirstLineHasColumnNames = true,
+            IgnoreUnknownColumns = true
+        };
+        CsvContext csvContext = new CsvContext();
+        StreamReader streamReader = new StreamReader(csvFile.OpenReadStream());
+        IEnumerable<TenantUploadModel> list = csvContext.Read<TenantUploadModel>(streamReader, csvFileDescription);
+        
+        List<string> flashTypes = flashResponse.flashTypes, flashMessages = flashResponse.flashMessages;
+        await using (var transaction = await _dbContext.Database.BeginTransactionAsync(IsolationLevel.Serializable))
+        {
+            foreach (var tenant in _dbContext.Tenants)
+            {
+                _dbContext.Tenants.Remove(tenant);
+            }
+            
+            foreach (TenantUploadModel tenant in list)
+            {
+                if (tenant.Name == null)
+                {
+                    _logger.LogWarning($"Name is missing from record {tenant.Email}.");

```

Comment:
Might be worth logging out the line number too as not all tenants will have emails